src/threading.ts

```diff
@@ -117,12 +118,33 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(f: Py_FrameObject): void {
```

Comment:
Also, ```throwException``` doesn't currently take an exception argument, as it is called inside ```f.raise_exception_here``` which already pushes the exception object on the frame stack. https://github.com/Mdrbhatti/Ninia/blob/master/src/frameobject.ts#L150